sdk/monitor/monitor-opentelemetry/src/metrics/quickpulse/filtering.ts

```diff
@@ -0,0 +1,561 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import {
+  DerivedMetricInfo,
+  KnownTelemetryType,
+  FilterInfo,
+  KnownPredicateType,
+  CollectionConfigurationError,
+  FilterConjunctionGroupInfo,
+  KnownAggregationType,
+} from "../../generated";
+import { RequestData, TelemetryData, DependencyData, ExceptionData, TraceData } from "./types";
+import {
+  isRequestData,
+  isDependencyData,
+  isExceptionData,
+  isTraceData,
+  getMsFromFilterTimestampString,
+} from "./utils";
+
+export class TelemetryTypeError extends Error {
+  constructor(message: string) {
+    super(message);
+    this.name = "TelemetryTypeError";
+  }
+}
+
+export class UnexpectedFilterCreateError extends Error {
+  constructor(message: string) {
+    super(message);
+    this.name = "UnexpectedFilterCreateError";
+  }
+}
+
+export class DuplicateMetricIdError extends Error {
+  constructor(message: string) {
+    super(message);
+    this.name = "DuplicateMetricIdError";
+  }
+}
+export class MetricFailureToCreateError extends Error {
+  constructor(message: string) {
+    super(message);
+    this.name = "MetricFailureToCreateError";
+  }
+}
+
+export enum KnownRequestColumns {
+  Url = "Url",
+  Duration = "Duration",
+  ResponseCode = "ResponseCode",
+  Success = "Success",
+  Name = "Name",
+}
+
+export enum KnownDependencyColumns {
+  Target = "Target",
+  Duration = "Duration",
+  Success = "Success",
+  Name = "Name",
+  ResultCode = "ResultCode",
+  Type = "Type",
+  Data = "Data",
+}
+
+const knownStringColumns = new Set<string>([
+  KnownRequestColumns.Url,
+  KnownRequestColumns.Name,
+  KnownDependencyColumns.Target,
+  KnownDependencyColumns.Type,
+  KnownDependencyColumns.Data,
+  "Message",
+  "Exception.Message",
+  "Exception.StackTrace",
+]);
+
+export class Validator {
```

Comment:
Calling this FilterValidator and putting it in its own ts file would be great, filtering.ts is a little vague and currently have to much code, same for other classes defined here, Projection, Filter, etc. You can also create a filtering folder and just leave it as Validator, then you can have all other classes with their own files in there